alphafold3_pytorch/alphafold3.py

```diff
@@ -2619,6 +2638,18 @@ def forward(
 
         pairwise_init = pairwise_init + relative_position_encoding
 
+        # Handle token_bonds embedding
+        if token_bonds is None:
+            # Default to a single chain if token_bonds is not provided
+            token_bonds = torch.eye(pairwise_init.shape[1], dtype=pairwise_init.dtype, device=pairwise_init.device)
+            token_bonds = token_bonds.unsqueeze(0).expand(pairwise_init.shape[0], -1, -1)
+        else:
+            token_bonds = token_bonds.to(pairwise_init.dtype).to(pairwise_init.device)
```

Comment:
i think device would be taken care of externally by the trainer, and the typing would take care of the dtype